### DISCUSSION TITLE
ROG Flow Z13 Additional Features

### DISCUSSION THREAD

**Discussion by CommanderGizmo** [seerge/g-helper#2521] [Ideas]
I have a ROG Flow Z13 (GZ301ZE).  The G-Helper app is really awesome and does a great job making the Amoury Crate features actually accessible in a easy and fast package.  Thank you for making it!

However the Armoury Crate package on the Z13 offers a few features and options that I use frequently, but G-Helper doesn't have.  I also have found a way to extend the battery life out to up to 9 hours from the 6 it usually gets.  However, G-Helper doesn't have a convenient way to deploy it.  So, here are a few features I would love to see in G-Helper.

### 1) Separate Lighting on DC/Battery 
The AC app has the option to disable a lighting zone when on battery.  This can save a lot of precious power on a device as battery abusive as the Z13.  I'd like to see the "Awake" option for lighting be changed to two separate options "DC Awake" and "Battery Awake".

### 2) Long Battery Power Profile
The Z13, as mentioned, is extremely power hungry.  I have created an advanced power profile for extending the battery by 50%.  However, I can't just replace the "silent" power profile with it.  Sometimes you want passive, silent cooling but need more power than the long battery option would allow.  Other times, you just want it silent when plugged in.  Thus, an additional option would be needed when you want to extend the battery.  Think of it as Silent Plus.  _It would be nice to have an option to automatically switch to this when DC is removed._

### 3) Turbo Mode Detection
The Z13 turbo mode unlocks the normal 65W limit for the laptop and allows it to run at 100W.  You can only use Turbo mode if an official Asus power adapter is connected.  PD or USB-C chargers only allow the use of Performance Mode (I think G-Helper calls this Balanced) at 65W.  It would be nice if G-Helper detected the ability to go from 65W to 100W and it was clear that is what happened when enabling Turbo mode.  As it happens, I'm not sure if Turbo is even enabling the extra wattage at all.

Thank you for considering the addition of these features.  I'm happy to do debug testing or offer other info or assistance in adding these features it would be helpful.

Thank you!

**Comment by seerge** [comment]
@CommanderGizmo hello,

Everything that you’re asking is already possible 

1. App remembers backlight brightness per battery/plugged and if switches automatically 

2. You can already customise Silent mode to your needs. Last used mode is remembered per batter/plugged and switched automatically as well

3. G-Helper doesn’t require “official” charger to enable Turbo mode. You can enable it any time.

Keep in mind that BIOS may still have restrictions for Turbo if there is not enough power from charger tho. Its handled by hardware

**Reply by CommanderGizmo** [reply]
@seerge 

Sorry, I must not have explained what I was trying to say very well.  These features simply are not present or possible at this time.

  1. The backlight brightness is not what I was referring to.  I was referring to the checkboxes for different lighting regions under the brightness slider.  In battery mode, it is helpful to have the keyboard backlight enabled, but not the Z13's lighting bar (a panel on the back).  However, this should be turned on when DC is connected.  In my testing, the app does not remember the state of the checkboxes DC vs Battery.  Armoury Crate does allow these to be set individually as an on/off checkbox per DC/Battery states.

  2. I know I can customize Silent mode, and already have.  However, as stated, I need to keep Silent mode as is and have another power option beyond that.  On the Z13, silent is very handy even when on AC for light work as it keeps it quiet and allows it to stay cool before putting it into your bag.  However, when on battery, you need to switch to another profile that severely throttles the P-Cores to conserve battery.  This is too slow for some things though, so switching back and forth from Silent to Extreme Battery Saver is needed to extend the battery run time.  This would be extremely cumbersome to edit the Silent mode each time you want to make a change.

  3. I realize that G-Helper doesn't require the official charger to enable Turbo, but the BIOS does.  Putting it into Turbo in G-Helper doesn't appear to run the CPU at full speed as expected.  Thus, it would be nice if it actually detected if the switch to Turbo was successful, or let the user know when it isn't really enabled.  Your explanation is actually the very problem I was trying to describe.  It doesn't require the adapter to _appear_ to enable the mode, but then it doesn't actually get enabled in BIOS and nothing happens.

Does that make more sense?  Hopefully I am coming across clearly.  Sorry for the confusion.  I'm happy to answer any questions or clarify any confusing language.

**Reply by seerge** [reply]
@CommanderGizmo hello

1. I don’t plan to over complicate things with extra checkboxes as it will conflict with current way app remembers backlight brightness per plugged/battery states and may confuse people. If lightning bar bothers you - you can just turn it off all together :)

2. You can create a custom mode based on Silent already. Just select Silent and click (+) on top right of Fans + Power. It will use Silent as a base for new profile.

3. Turbo mode is set even on unofficial charger. If BIOS restrict it somehow, then it’s how device is programmed. When you will plug in official charger - app would re-apply Turbo mode again and it will be unrestricted. There is no need to change anything here additionally.

There are no known to me ways to distinguish official adapter from a generic one

**Reply by CommanderGizmo** [reply]
@seerge Thank you for the great explanation.

   1.  I understand your decision.  Perhaps I'll make a fork with the options I want.

   2.  How did I miss that option?  Well, I appreciate you explaining it to me.  I read the documentation and still missed it.  Thanks, I'll use that feature from now on.

   3. So, the program will do a check each time the power state changes?  That makes sense and should be sufficient to know whether Turbo mode is really working or not.  Thank you for helping me understand how the feature works.

Thank you for the great replacement for Armoury Crate.  It really is a fantastic app to have.

**Reply by seerge** [reply]
@CommanderGizmo 

> So, the program will do a check each time the power state changes?

G-Helper re-applies last remembers mode every time power delivery source changes (i.e. you plug / unplug).

So if you used Turbo last time you were plugged in (to usb-c as well) and plug in official charger - Turbo will be set again.
